### ISSUE TITLE
python: Can't find libBTKCommon.so.0.4dev

### ISSUE THREAD

**Issue by kidzik** [opensim-org/opensim-core#1353]
After installing the current version on Ubuntu 16.04, installing python bindings and running 
`import opensim` in python I'm getting: 

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/opensim-4.0-py2.7.egg/opensim/__init__.py", line 2, in <module>
    from common import *
  File "/usr/local/lib/python2.7/dist-packages/opensim-4.0-py2.7.egg/opensim/common.py", line 28, in <module>
    _common = swig_import_helper()
  File "/usr/local/lib/python2.7/dist-packages/opensim-4.0-py2.7.egg/opensim/common.py", line 24, in swig_import_helper
    _mod = imp.load_module('_common', fp, pathname, description)
ImportError: libBTKCommon.so.0.4dev: cannot open shared object file: No such file or directory
```

This

```
sudo cp opensim_dependencies_install/BTK/lib/btk-0.4dev/* /usr/lib
```

solves the problem but maybe I should just set up some path? In any case it seems it's not documented (I will try to fix such documentation issues by myself as soon as I get through the documentation by myself and understand the architecture better :).

**Comment by klshrinidhi** [comment]
Please post how you built OpenSim ( all the commands you used ) so I can reproduce the problem.

**Comment by kidzik** [comment]
I followed the "impatient" path:
```
sudo apt-get update
sudo apt-get --yes install git cmake cmake-curses-gui \
                           freeglut3-dev libxi-dev libxmu-dev \
                           liblapack-dev swig python-dev \
                           openjdk-8-jdk
export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
git clone https://github.com/opensim-org/opensim-core.git
mkdir opensim_dependencies_build
cd opensim_dependencies_build
cmake ../opensim-core/dependencies/ \
      -DCMAKE_INSTALL_PREFIX='~/opensim_dependencies_install' \
      -DCMAKE_BUILD_TYPE=RelWithDebInfo
make -j8
cd ..
mkdir opensim_build
cd opensim_build
cmake ../opensim-core \
      -DCMAKE_INSTALL_PREFIX="~/opensim_install" \
      -DCMAKE_BUILD_TYPE=RelWithDebInfo \
      -DOPENSIM_DEPENDENCIES_DIR="~/opensim_dependencies_install" \
      -DBUILD_PYTHON_WRAPPING=ON \
      -DBUILD_JAVA_WRAPPING=ON \
      -DWITH_BTK=ON
make -j8
ctest -j8
```
and then just
```
cd opensim_install/lib/python2.7/site-packages/
python setup.py install
```

**Comment by chrisdembia** [comment]
@klshrinidhi I like your approach to a fix in #1354 (using RPATH). I want to share some other thoughts though.

1. At least for BTK, we should be able to statically link, since BTK is not exposed in our interface (only included in CPP files). (side note: I know docopt is currently dynamically linked, but I am hoping to make a future PR in which it is only statically linked). Regardless of if we dynamically or statically link for BTK, we will still have this issue for Simbody, so this is not really a solution, but could still help.

2. It is somewhat of an expected workflow on Linux to either put your .so files in a system path like `/usr/lib` or `/usr/local/lib` or to otherwise set LD_LIBRARY_PATH. But it is not obvious from our README that there are .so files that need to be on your path. So we could add such a statement to the README. In particular, [this part](https://github.com/opensim-org/opensim-core#set-environment-variables-2) of the README is out of date. Also, the README could document the `OPENSIM_NO_LIBRARY_PATH_ENV_VAR` CMake variable.

3. Right now, we copy Simbody .so files into OpenSim's lib directory. This isn't so great, and we should avoid it if possible.

**Comment by klshrinidhi** [comment]
Addressed by #1354.
